addons/csi/vsphere/csiController.yaml

```diff
@@ -0,0 +1,218 @@
+# Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{ if .Cluster.Features.Has "externalCloudProvider" }}
+
+{{ if eq .Cluster.CloudProviderName "vsphere" }}
+{{ $version := "UNSUPPORTED" }}
+{{ if eq .Cluster.MajorMinorVersion "1.18" }}
+{{ $version = "v2.2.1 "}}
+{{ end }}
+{{ if eq .Cluster.MajorMinorVersion "1.19" }}
+{{ $version = "v2.2.1 "}}
+{{ end }}
+{{ if eq .Cluster.MajorMinorVersion "1.20" }}
+{{ $version = "v2.2.1 "}}
+{{ end }}
```

Comment:
What about the latest Kubernetes version (e.g. currently 1.21)? If someone would create a cluster running 1.20 and upgrade it to 1.21, they would lose the CSI. I'm aware that the [compatibility matrix](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/docs/book/releases/v2.2.1.md#kubernetes-release) for the CSI driver says that the maximum supported version is 1.20, but we should somehow handle this case.

I think it should be safe to use CSI v2.2.1 with 1.21 as well, but we should test it.

Reply:
I'll test it and if it works I'll set CSI v2.2.1 for 1.21 as well

Reply:
I made extensive testing and k8s 1.21 is actually well supported, gonna add it.